### DISCUSSION TITLE
Could this run as a web app, so it can be installed and previously downloaded roms be playable offline?

### DISCUSSION THREAD

**Discussion by ghost** [EmulatorJS/EmulatorJS#890] [Ideas]
I just figure that would be a useful feature.

**Comment by ethanaobrien** [comment]
This isn't one of the goals of the emulatorjs project. 

EmulatorJS is intended to be embedded and used in a bigger application, like how projects like gaseous and romm work. (Or just the adding it to your website)

The webapp idea isn't one of the goals of this project, but if you or anybody else wants to do this using EmulatorJS as the back end it should be relatively simple

**Comment by ghost** [comment]
Oh ok, I wasn't aware Gaseous or romm existed. Looks like romm is closer to what I was looking for, and plans to be more like what I originally wanted.

**Reply by michael-j-green** [reply]
> Oh ok, I wasn't aware Gaseous or romm existed. Looks like romm is closer to what I was looking for, and plans to be more like what I originally wanted.

Gaseous developer here. I just wanted to ask what feature Romm has that makes it closer to what you're looking for? Gaseous supports importing a directory full of existing ROMs, but as far as I'm aware neither supports offline play. That is, both need to have a server running on your own host.

I do have full offline mode as a long term goal, but there's a significant amount of work to get there from where I am at the moment.

**Reply by ghost** [reply]
Frankly I still don't really know what the differences are between romm and Gaseous as I don't have time to test and deploy, but the romm GitHub has more activity and interactions and a website and so on which to me indicates they are further along in development and have a higher capacity to resolve issues that prop up and develop new features. 

I could be wrong though. Ive only really thumbed through the repository and website.

How hard would it be to develop a plugin to something like Retroarch that lets you download ROMs and bioses from within the interface and backup/sync game saves?

**Reply by michael-j-green** [reply]
That's fair. Gaseous is only a year or so old. I think Romm is slightly older. Though I'm a one man shop (though if you know C# or Javascript, you're welcome to contribute! :) ).

That feature you're asking for is what both projects are able to do in browser - but if you wanted to use native RetroArch, I don't see why couldn't build a locally installable app to sync saves and ROMs with the server when the network is available. Gaseous (and I imagine Romm too) has an API you can work with.

Indeed another long term goal of mine is to have a locally installable app to work alot like Steam does (which is the inspiration for the Gaseous name).
